src/formatDate.js

```diff
@@ -8,7 +8,57 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromValue = fromFormat[fromFormat.length - 1];
+
+  const toValue = toFormat[toFormat.length - 1];
+
+  const newDate = date.split(fromValue);
+
+  const newArray = [];
+
+  const dateObject = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObject[fromFormat[i]] = newDate[i];
+  }
+
+  const indexOfYear = fromFormat.includes('YYYY')
+    ? fromFormat.indexOf('YYYY')
+    : fromFormat.indexOf('YY');
+
+  if (!('YYYY' in dateObject)) {
+    if (newDate[indexOfYear] >= 30) {
+      dateObject.YYYY = '19' + newDate[indexOfYear];
+    }
+
+    if (newDate[indexOfYear] < 30) {
+      dateObject.YYYY = '20' + newDate[indexOfYear];
+    }
+  }
```

Comment:
The conversion of two-digit years to four-digit years assumes that years >= 30 belong to the 1900s and < 30 to the 2000s. This logic might not be universally applicable and could lead to incorrect results. Consider making this logic configurable or adding a comment to clarify this assumption.